packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2886,6 +2889,9 @@ export class ContainerRuntime
 			// Cleanup wip summary in case of failure
 			this.summarizerNode.clearSummary();
 
+			// ! This needs to happen before we resume inbound queues to ensure heuristics are tracked correctly
+			this._summarizer?.recordSummaryAttempt?.(summaryRefSeqNum);
```

Comment:
If submitSummary fails, at attempt won't be noted anymore right?

Reply:
Wait no, this is in finally block so it will get executed even if the summary attempt fails right. Unless it fails before the try block.
Can you verify this in tests? With you last update, record attempt may be called twice (not sure that impact it will have though).